// ripple-notify/ripple.js

/*
  Copyrights: Codiify
  Website: https://www.codiify.com
  Developer: Md. Rubel Rana
  Dev website: https://www.rubelrana.dev

  Official x account: https://x.com/codiify_ui
  Developer x account: https://www.x.com/rubel_codiify
 */
 
export const RippleConfig = {
    timeout: 3000,
    animationSpeed: '0.5s',
    duration: 500,
    colorful: true,
    icon: true,
    position: 'bottom-right',
    colors: {
        success: {
            backgroundColor: '#ECFDF3',
            color: '#008A2E',
            borderColor: '#D3FDE5'
        },
        error: {
            backgroundColor: '#FFF0F0',
            color: '#E60000',
            borderColor: '#FFE0E1'
        },
        info: {
            backgroundColor: '#F0F8FF',
            color: '#0973DC',
            borderColor: '#D3E0FD'
        },
        warning: {
            backgroundColor: '#FFFCF0',
            color: '#DC7609',
            borderColor: '#FDF5D3'
        },
        default: {
            backgroundColor: '#FFFFFF',
            color: '#171717',
            borderColor: '#EDEDED'
        }
    }
};
